code/game/machinery/tcomms/relay.dm

```diff
@@ -26,6 +28,11 @@
 /obj/machinery/tcomms/relay/Initialize(mapload)
 	. = ..()
 	component_parts += new /obj/item/circuitboard/tcomms/relay(null)
+	if(check_power_on())
```

Comment:
Mostly dupe code here as well. Best to check if you can put this a level up

Reply:
This might be cleanable but I aint sure of it